### PR TITLE
docs: Update external workloads instructions

### PR DESCRIPTION
**Pull request by michi-covalent**

### CODE REVIEW

File: Documentation/network/external-workloads.rst [hunk 1]
```diff
@@ -201,7 +186,7 @@ Next you can check the status of the Cilium agent in your external workload:
 
 .. code-block:: shell-session
 
-    cilium status
+    sudo cilium status
```
Comment (michi-covalent) [inline Documentation/network/external-workloads.rst:193]: wasn't sure how to document this. in my test setup i needed to do `sudo` to connect to /var/run/cilium/cilium.sock, but i guess it depends on how you log in to your external workload 💭

File: Documentation/network/external-workloads.rst [hunk 2]
```diff
@@ -72,23 +72,8 @@ clustermesh enable`` after ``cilium install``:
 
 .. code-block:: shell-session
 
-    cilium install --config tunnel-protocol=vxlan
-    cilium clustermesh enable
-
-Config option ``tunnel-protocol=vxlan`` overrides any default that could
-otherwise be auto-detected for your k8s cluster. This is currently a
-requirement for external workload support.
```
Comment (giorio94) [inline Documentation/network/external-workloads.rst:80]: Should this be preserved?

**Review by giorio94 — APPROVED** [review]

File: Documentation/network/external-workloads.rst (hunk 1 above)
Comment (qmonnet) [reply Documentation/network/external-workloads.rst:193]: I'm not sure we have any particular convention, this looks acceptable to me.

**Review by qmonnet — APPROVED** [review]
Looks good once Marco's feedback is sorted.

Same comment as on https://github.com/cilium/cilium/pull/26608#issuecomment-1619833790, do you need a `needs-backport/1.14` label here?

File: Documentation/network/external-workloads.rst (hunk 2 above)
Comment (michi-covalent) [reply Documentation/network/external-workloads.rst:80]: yeah we can preserve it, i'll slightly rephrase it based on: https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/cilium-configmap.yaml#L439

> The ``tunnel=vxlan`` Helm value sets `tunnel-protocol` Cilium agent configuration option to `vxlan`. This is currently a requirement for external workload support.

Comment (michi-covalent) [reply Documentation/network/external-workloads.rst:80]: updated 📝

**Comment by qmonnet** [comment]
Doc-change only - reviews are in, and the relevant workflows passed successfully. We're good to merge this.

**Pull request merged**